### PR TITLE
ProjectionRepository automatic catch-up.

### PR DESCRIPTION
**Pull request by armiol**
Summary of changes:
* Turn on automatic catch-up for `ProjectionRepository` instances the storage is initialized. 
* Make this option configurable by introducing an alternative `ProjectionRepository` constructor.
* Add tests for the behaviour changes.

### CODE REVIEW

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

**Comment by armiol** [comment]
@alexander-yevsyukov the PR is now complete. Please feel free to review.

File: server/src/main/java/org/spine3/server/projection/ProjectionRepository.java [hunk 1]
```diff
@@ -123,6 +150,13 @@ protected Storage createStorage(StorageFactory factory) {
     public void initStorage(StorageFactory factory) {
         super.initStorage(factory);
         setStatus(Status.STORAGE_ASSIGNED);
+
+        if(catchUpAfterStorageInit) {
+            if(log().isDebugEnabled()) {
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/projection/ProjectionRepository.java]: Why do we check the level here? We do not pass parameters to the log.

Comment (armiol) [inline server/src/main/java/org/spine3/server/projection/ProjectionRepository.java]: I see your point. I have added some parameters to let a developer know which `ProjectionRepository` is catching up.

**Pull request merged**